File: Java/215.java
__________________________________________________________________________________________________
sample 0 ms submission
class Solution {
    static int findKthLargest(int[] nums, int k){
        int n = nums.length;
        partition(nums,0, n-1, k);
        return nums[k-1];
    }

    static void partition(int[] nums, int start, int end, int k){
        if(start >= end)
            return;

        int i = start;
        int j = end;
        int pivot = median3(nums, start, end);

        while(i <= j){
            while(nums[i] > pivot)
                i++;
            while(nums[j] < pivot)
                j--;
            if(i <= j){
                swap(nums, i, j);
                i++;
                j--;
            }
        }

        if(k <= i){
            partition(nums, start, i - 1, k);
        }else{
            partition(nums, i, end, k);
        }
    }

    static int median3(int[] nums, int left, int right){
        int center = (left + right) / 2;
        if(nums[left] > nums[center])
            swap(nums, left, center);
        if(nums[left] > nums[right])
            swap(nums, left, right);
        if(nums[center] > nums[right])
            swap(nums, center, right);

        return nums[center];
    }

    static void swap(int[] nums, int i, int j){
        int tmp = nums[j];
        nums[j] = nums[i];
        nums[i] = tmp;
    }
}
__________________________________________________________________________________________________
sample 34436 kb submission
class Solution {
    public int findKthLargest(int[] nums, int k) {
    PriorityQueue<Integer> pq=new PriorityQueue<Integer>((n1,n2)->n1-n2);
    for(int n:nums)
    {
        pq.add(n);
        if(pq.size()>k)pq.poll();
    }
        return pq.poll();
    }
}
__________________________________________________________________________________________________

File: Python3/215.py
__________________________________________________________________________________________________
sample 60 ms submission
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        ans = heapq.nlargest(k, nums)
        return ans[-1]
__________________________________________________________________________________________________
sample 13484 kb submission
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        
        def partition(start, end):
            pivot = start
            while start <= end:
                while start <= end and nums[start] <= nums[pivot]:
                    start += 1
                while start <= end and nums[end] > nums[pivot]:
                    end -= 1
                if start > end: break
                nums[start], nums[end] = nums[end], nums[start]
            nums[end], nums[pivot] = nums[pivot], nums[end]    
            return end
            
        end = len(nums) - 1
        distance = len(nums) - k
        start = 0
        while start < end:
            pivot = partition(start, end)
            # print(nums)
            if pivot < distance:
                start = pivot + 1
            elif pivot > distance:
                end = pivot - 1
            else:
                # print(nums)
                return nums[pivot]
        print(nums)
        return nums[start]
__________________________________________________________________________________________________
